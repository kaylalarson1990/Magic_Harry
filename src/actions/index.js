export const setCharacters = characters => ({
  type: "SET_CHARACTERS",
  characters
});

export const setHouses = houses => ({
  type: "SET_HOUSES",
  houses
});

export const setSpells = spells => ({
  type: "SET_SPELLS",
  spells
});
