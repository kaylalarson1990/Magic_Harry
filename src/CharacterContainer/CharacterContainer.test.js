import {
  CharacterContainer,
  mapStateToProps,
  mapDispatchToProps
} from "./CharacterContainer.jsx";
import { setCharacters } from "../actions/index";
import { shallow } from "enzyme";
import React from "react";

describe(CharacterContainer, () => {
  let wrapper, mockData;
  beforeEach(() => {
    mockData = [
      {
        id: "1",
        name: "Kayla",
        role: "student",
        house: "KlioPold",
        school: "Turing",
        species: "human"
      }
    ];

    wrapper = shallow(<CharacterContainer characters={mockData} />);
  });

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe("mapStateToProps", () => {
  it("should return an object with a characters array", () => {
    const mockState = {
      characters: "Kayla",
      type: "SET_CHARACTERS"
    };
    const expected = {
      characters: "Kayla"
    };
    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps).toEqual(expected);
  });
});

describe("mapDispatchToProps", () => {
  it("calls dispatch with a setCharacters action", () => {
    const mockDispatch = jest.fn();
    const actionToDispatch = setCharacters("Kayla");
    const mappedDispatch = mapDispatchToProps(mockDispatch);
    mappedDispatch.setCharacters("Kayla");
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });
});
