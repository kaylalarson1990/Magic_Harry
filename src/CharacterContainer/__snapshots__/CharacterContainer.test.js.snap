// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterContainer should match snapshot 1`] = `
<Fragment>
  <article>
    <h2
      className="title"
    >
      Harry Potter Characters
    </h2>
    <input
      class="search__input"
      placeholder="Search"
      type="text"
    />
  </article>
  <section
    className="characters"
  >
    <Characters
      house="KlioPold"
      key="1"
      name="Kayla"
      role="student"
      school="Turing"
      species="human"
    />
  </section>
</Fragment>
`;
